### PR TITLE
Blog app - Add forms

### PR DESCRIPTION
**Pull request by r-ahmed2022**
In this PR, I completed the following:-
- Created a method called current_user in your ApplicationController to make current user data available to all controllers.
-  Create forms to perform the following functions such as creating post on behalf of current user , or comment on behalf of current_user
image

image

- Allowed Users to add likes to Posts.
image

### CODE REVIEW

**Review by tresorsawasawa — COMMENTED** [review]
# STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @r-ahmed2022,:wave: :wave: 
While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 🛑 🛑

You have submitted a project that can't run because the below error when trying to run `migrations`. I suggest you first delete your older database, create a new one then you can fix the issue.
Screenshot from 2023-05-18 20-22-59 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

---
Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me [@tresorsawasawa](https://github.com/tresorsawasawa) when you comment so I can receive the notification or use slack with the same name._

---
_Invalid Code Review Request **Does not count** into the code reviews limit._

**Review by rloterh — COMMENTED** [review]
Hi Riyaz, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

There is a mismatch in database table name, and hence, an error is thrown in attempt to migrate the database during setup process. Kindly ensure to fix this issue.

image


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by r-ahmed2022** [comment]
> Hi Riyaz,
> 
> While you made a great effort in this project, unfortunately, I cannot proceed to review your code.
> 
> ## Invalid Code Review Request
> There is a mismatch in database table name, and hence, an error is thrown in attempt to migrate the database during setup process. Kindly ensure to fix this issue.
> 
> image
> 
> Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _Invalid Code Review Request **does not count** into the code reviews limit._

@rloterh , https://www.loom.com/share/ecdec9951ac24badac5864d461c45d3b

File: app/controllers/likes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class LikesController < ApplicationController
+  def create
+    @like = Like.new
+    @like.author = current_user
+    @like.post_id = params[:post_id]
+
+    if @like.save
+      flash[:success] = 'You liked this post!'
+      redirect_to "/users/#{params[:author_id]}/posts/#{params[:post_id]}"
```
Comment (Rustamxon7) [inline app/controllers/likes_controller.rb:9]: **Well-implemented code!**

- [ ] However, your redirect_to approach for likes leades me to broken link. To fix this issue you can do like that:

```ruby
      redirect_to user_post_comments_path(current_user, @comment.post_id)

``` 

Kindly fix the issue and resubmit it again :)

**Review by Rustamxon7 — CHANGES_REQUESTED** [review]
Hello @r-ahmed2022,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_
_Great readme_
_Well-implemented code_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/likes_controller.rb (hunk 1 above)
Comment (r-ahmed2022) [reply app/controllers/likes_controller.rb]: Done, as advised

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand: :raising_hand: @r-ahmed2022, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :green_circle: 

- PR is professional :+1: 
- Added all requested forms :+1: 
- Linter checks passed :heavy_check_mark: 

## Optional suggestions
**N/A**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**